docs/state_delta_resolution.tex

```diff
@@ -0,0 +1,168 @@
+\newtheorem{lemma}[theorem]{Lemma}
+\newtheorem{corollary}[theorem]{Corollary}
+
+\setlength{\parskip}{1em}
+
+\newcommand{\inlinemaketitle}{{\let\newpage\relax\maketitle}}
+
+
+\begin{document}
+	
+\maketitle
+	
+We first define some basic functions that correspond to the synapse code.
+
+Let $K$ be the set of all type/state key tuples and $E$ the set of all events. We can then define arbitrary functions:
+\begin{equation}
+\begin{split}
+f: F \rightarrow E\\
+g: G \rightarrow E\\
+\end{split}
+\end{equation} which we call state maps, for $F, G \subset K$.
+
+We can then compute the set of all ``unconflicted events":
+\begin{equation}
+U_{f,g} = \{x\ |\ \forall x \in F \cap G,\ f(x) = g(x)\} \cup (F \bigtriangleup G)
+\end{equation} i.e. the set of state keys where $f$ and $g$ don't conflict. Similarly, we define:
+\begin{equation}
+\begin{split}
+u_{f,g} : U_{f,g} \longrightarrow &\ E\\
+x \longmapsto &\ \begin{cases}
+	f(x), & \text{if}\ f \in F \\
+	g(x), & \text{otherwise}
+\end{cases}
+\end{split}
+\end{equation} which gets the unconflicted event for a given state key.
+
+
+We can also define a function on $C_{f,g} = F \cup G \setminus U_{f,g}$:
+\begin{equation}
+c_{f,g}: C_{f,g} \rightarrow E
+\end{equation} which is used to resolve conflicts between $f$ and $g$. Note that $c_{f,g}$ is either $f(x)$ or $g(x)$. 
+
+Now we define:
+\begin{equation} 
+\begin{split}
+r_{f,g}: F \cup G \longrightarrow &\ E\\
+x \longmapsto &\ \begin{cases}
+u_{f,g}(x), & \text{if}\ x \in U_{f,g}\\
+c_{f,g}(x), & \text{otherwise}
+\end{cases} 
+\end{split}
+\end{equation} which we call the resolved state of $f$ and $g$.
+
+\begin{lemma}
+	$\forall x \in U_{f,g} \ s.t.\ g(x) = g'(x)$ then $r_{f,g}(x) = r_{f,g'}(x)$
+\end{lemma}
+
+We define 
+\begin{equation}
+\alpha: E \rightarrow \mathbb{P}(K)
+\end{equation} to be the mapping of an event to the type/state keys needed to auth the event, and
+\begin{equation}
+\alpha_{f,g}(x) = \alpha f(x) \cup \alpha g(x)
+\end{equation} which is the set of auth events required for $f(x)$ and $g(x)$. Note that $\alpha_{f,g}(x) \subset F \cup G$.
+
+Further, we can define
+\begin{equation}
+a_{f,g}(x) = \bigcup_{n=0}^\infty (\alpha_{f,g})^n(x)
+\end{equation} to be the auth chain of $f(x)$ and $g(x)$. This is well defined as there are a finite number of elements in $F \cup G$ and $a_{f,g} \rightarrow F \cup G$.
+
+If we consider the implementation of $c_{f,g}$ in Synapse we can see that it depends not only on the values of $x$, but also on the resolved state of their auth events, i.e. $r_{f,g}(\alpha_{f,g}(x))$. By ``depends on" we mean that if those are the same for different values of $f$ and $g$, then the result of $c_{f,g}(x)$ is the same.
+
+\begin{lemma}
+	 $c_{f,g}$ depends only on $a_{f,g}(x)$
+\end{lemma}
+
+\begin{proof}
+	$c_{f,g}(x)$ depends on $x \in a_{f,g}(x)$, and $r_{f,g}(\alpha_{f,g}(x))$. Now:
+	\[
+		\begin{split}
+		r_{f,g}(\alpha_{f,g}(x))\ =\ & u_{f,g}(\alpha_{f,g}(x)) \cup c_{f,g}(\alpha_{f,g}(x))\\
```

Comment:
u is only defined over the unconflicted part of the state key space. I think it's clear what you mean, but it might be good to be more precise here.